src/ssmDocument/commands/publishDocument.ts

```diff
@@ -0,0 +1,191 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { SSM } from 'aws-sdk'
+import * as vscode from 'vscode'
+import * as nls from 'vscode-nls'
+const localize = nls.loadMessageBundle()
+import { AwsContext } from '../../shared/awsContext'
+import { SsmDocumentClient } from '../../shared/clients/ssmDocumentClient'
+import { ssmJson, ssmYaml } from '../../shared/constants'
+import { ext } from '../../shared/extensionGlobals'
+import { getLogger, Logger } from '../../shared/logger'
+import {
+    DefaultPublishSSMDocumentWizardContext,
+    PublishSSMDocumentWizard,
+    PublishSSMDocumentWizardContext,
+    PublishSSMDocumentWizardResponse,
+} from '../wizards/publishDocumentWizard'
+import { stringify } from 'querystring'
+import * as telemetry from '../../shared/telemetry/telemetry'
+
+const DEFAULT_REGION: string = 'us-east-1'
+
+export async function publishSSMDocument(awsContext: AwsContext, outputChannel: vscode.OutputChannel): Promise<void> {
+    const logger: Logger = getLogger()
+
+    const textDocument = vscode.window.activeTextEditor?.document
+    if (!textDocument) {
+        let error = new Error('Could not get active text editor for local SSM Document definition')
+        logger.error(error.message)
+        throw error
+    }
+
+    if (textDocument.languageId !== ssmJson && textDocument.languageId !== ssmYaml) {
+        let supportedFormats = [ssmJson, ssmYaml]
+        let error = new Error(
```

Comment:
Ensure any messages you intend to surface to the user are localized and presented appropriately (example: showErrorMessage).
Only errors that are beyond our control should be thrown for VS Code to handle.